### ISSUE TITLE
Roles endpoint: Supply translated role titles

### ISSUE THREAD

**Issue by lukasgraf** [plone/plone.restapi#536]
The `@role` endpoint should also supply translated role titles, in addition to their ID.

This is needed for rendering the sharing page.

Example output:

```
[
  {
    "@id": "http://localhost:55001/plone/@roles/Contributor", 
    "@type": "role", 
    "id": "Contributor",
    "title": "Hinzufügen"
  }, 
  {
    "@id": "http://localhost:55001/plone/@roles/Editor", 
    "@type": "role", 
    "id": "Editor",
    "title": "Bearbeiten"
  },
# ...
]
```

We should be able to do this by quering for [`ISharingPageRole` utilities and returning their title](https://github.com/plone/plone.app.workflow/blob/6c548ac4a26706503df17279f1ea8d9b0b7ba099/plone/app/workflow/browser/sharing.py#L123-L148).

These can then be referenced from the `@sharing` endpoint via their `@id`.

**Comment by erral** [comment]
This is somehow also related to #337 where we need to expose the translated titles for Content-types and so.

**Comment by erral** [comment]
We just need to add the 'title' property and wrap the title into a zope.i18n.translate call to get it translated.

**Comment by lukasgraf** [comment]
This has been implemented in #545.
